aspnet/mvc/views/razor.rst

```diff
@@ -1,8 +1,746 @@
+
+.. code-block:: none
+
+  <p>Last week this time: @(DateTime.Now - TimeSpan.FromDays(7))</p>
+
+Any content within the @() parenthesis is evaluated and rendered to the output.
+
+Expression encoding
+-------------------
+
+Non-:dn:iface:`~Microsoft.AspNetCore.Html.IHtmlContent` content is HTML encoded. For example, the following Razor markup:
+
+.. code-block:: none
+
+  @("<span>Hello World</span>")
+
+Renders this HTML:
+
+.. code-block:: none
+
+  &lt;span&gt;Hello World&lt;/span&gt;
+
+Which the browser renders as:
+
+``<span>Hello World</span>``
+
+:dn:cls:`~Microsoft.AspNetCore.Mvc.ViewFeatures.HtmlHelper` :dn:method:`~Microsoft.AspNetCore.Mvc.ViewFeatures.HtmlHelper.Raw` output is not encoded but rendered as HTML markup. :dn:cls:`~Microsoft.AspNetCore.Mvc.ViewFeatures.HtmlHelper` implements :dn:iface:`~Microsoft.AspNetCore.Html.IHtmlContent`
+
+.. warning:: Using ``HtmlHelper.Raw`` on unsanitzed user input is a security risk. User input might contain malicious JavaScript or other exploits. Sanitizing user input is difficult, avoid using ``HtmlHelper.Raw`` on user input.
+
+The following Razor markup:
+
+.. code-block:: none
+
+  @Html.Raw("<span>Hello World</span>")
+
+Renders this HTML:
+
+.. code-block:: none
+
+  <span>Hello World</span>
+
+Which the browser renders as:
+``Hello World``
+
+Razor code blocks
+------------------
+
+Razor code blocks start with ``@`` and are enclosed by ``{}``. Unlike expressions, C# code inside code blocks is not rendered.
+
+.. comment: sample shows var defined in first code block available in 2nd block. Some customer confusion on that.
+
+.. literalinclude:: razor/sample/Views/Home/Contact.cshtml
+  :language: html
+  :start-after: }
+  :end-before: @* End of greeting *@
+
+Renders this HTML markup:
+
+.. code-block:: none
+
+  <!-- Single statement blocks.  -->
+
+  <!-- Multi-statement block.  -->
+      <p>The greeting is :<br /> Welcome! Today is: Tuesday -Leap year: True</p>
+      <p>The value of your account is: 7 </p>
+
+Implicit transitions
+^^^^^^^^^^^^^^^^^^^^^
+
+The default languge in a code block is C#, but you can transition back to HTML. HTML within a code block will transition back into rendering HTML:
+
+.. code-block:: none
+
+  @{
+      var inCSharp = true;
+      <p>Now in HTML, was in C# @inCSharp</p>
+  }
+
+Explicit delimited transition
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To define a sub-section of a code block that should render HTML, surround the characters to be rendered with the ``<text>`` tag:
+
+.. code-block:: none
+
+  @{
+  /* C# */<text>I'm HTML</text>/* C# */
+  }
+
+Which renders ``I'm HTML`` without the ``text`` tags. You generally use this approach when you want to render HTML that is not surrounded by an HTML tag.
+
+.. _Explicit-Line-Transition-label:
+
+Explicit Line Transition with ``@:``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To render an entire line inside of a code block, utilize the ``@:`` characters:
```

Comment:
Also should we explain `@:` as being equivalent to putting <text> tags around the rest of the line?

In other words, is this:

``
/* Still C# */@: <p>Hello World</p> /* This is not C#, it's HTML */
```

Equivalent to this?:

```
/* Still C# */<text> <p>Hello World</p> /* This is not C#, it's HTML */</text>
```